src/main/java/com/monumental/services/UserService.java

```diff
@@ -31,9 +50,36 @@ public User signup(CreateUserRequest userRequest) throws InvalidEmailOrPasswordE
         user.setRole(Role.COLLABORATOR);
         user.setFirstName(userRequest.getFirstName());
         user.setLastName(userRequest.getLastName());
-        // TODO: Disable once email verification is in place
         user.setIsEnabled(true);
+        user.setIsEmailVerified(false);
         this.userRepository.save(user);
+
+        this.sendVerificationEmail(user, this.generateVerificationToken(user), appUrl, locale);
+
         return user;
     }
+
+    public VerificationToken generateVerificationToken(User user) {
+        String token = UUID.randomUUID().toString();
+        VerificationToken verificationToken = new VerificationToken();
+        verificationToken.setToken(token);
+        verificationToken.setUser(user);
+        this.tokenRepository.save(verificationToken);
+        return verificationToken;
+    }
+
+    public void sendVerificationEmail(User user, VerificationToken token, String appUrl, Locale locale) {
+        String recipientAddress = user.getEmail();
+        String subject = "Registration Confirmation";
+        String confirmationUrl
+                = appUrl + "/signup/confirm?token=" + token.getToken();
+        String message = this.messages.getMessage("registration.success", null, locale);
+
+        SimpleMailMessage email = new SimpleMailMessage();
+        email.setTo(recipientAddress);
+        email.setFrom("noreply@monuments.us.org");
```

Comment:
Can we move `"noreploy@monuments.us.org"` to a constant?

Reply:
I made it an environment variable that defaults to `noreply@monuments.us.org`, in case we ever need to use a different one for local development or something.